packages/broker/src/plugins/mqtt/MqttServer.ts

```diff
@@ -1,303 +1,93 @@
-                connection.id, connection.streamsAsString()
-            )
-
-            connection.client.suback({
-                granted: [packet.qos], messageId: packet.messageId
-            })
-        } catch (err) {
-            logger.debug(
-                'handleSubscribeRequest: socket "%s" failed to subscribe to stream "%s:%d" because of "%o"',
-                connection.id, topic, 0, err
-            )
-            connection.sendConnectionNotAuthorized()
+    publish(topic: string, payload: string) {
+        const packet: aedes.PublishPacket = {
+            topic,
+            payload,
+            cmd: MqttServer.COMMAND_PUBLISH as any,
+            qos: MqttServer.QOS_EXACTLY_ONCE as any,
+            dup: false,
+            retain: false
         }
-    }
-
-    _closeConnection(connection: Todo) {
-        this.connections.delete(connection)
-        logger.debug('closing client "%s" on streams "%o"', connection.id, connection.streamsAsString())
-
-        // Unsubscribe from all streams
-        connection.forEachStream((stream: Todo) => {
-            const object = {
-                messageId: 0,
-                unsubscriptions: [stream.getName()]
-            }
-
-            connection.sendUnsubscribe(object)
-
-            const streamObj = this.streams.get(stream.getId(), stream.getPartition())
-
-            if (streamObj) {
-                streamObj.removeConnection(connection)
-
-                if (streamObj.getConnections().length === 0) {
-                    logger.debug(
-                        'checkRoomEmpty: stream "%s:%d" is empty. Unsubscribing from NetworkNode.',
-                        stream.getId(), stream.getPartition()
-                    )
-
-                    this.subscriptionManager.unsubscribe(stream.getId(), stream.getPartition())
-                    this.streams.delete(stream.getId(), stream.getPartition())
-                }
+        this.aedes.publish(packet, (error?: Error) => {
+            if (error) {
+                logger.warn(`Publish error: ${error}`)
             }
         })
-
-        connection.close()
     }
 
-    broadcastMessage(streamMessage: Todo) {
-        const streamId = streamMessage.getStreamId()
-        const streamPartition = streamMessage.getStreamPartition()
-        const stream = this.streams.get(streamId, 0)
-
-        if (stream) {
-            const object = {
-                cmd: 'publish',
-                topic: stream.name,
-                payload: JSON.stringify(streamMessage.getParsedContent())
+    private static createAuthenicationHandler(apiAuthenticator: ApiAuthenticator): aedes.AuthenticateHandler {
+        return (_client: aedes.Client, _username: Readonly<string>|undefined, password: Readonly<Buffer>|undefined, done: (error: aedes.AuthenticateError|null, success: boolean|null) => void) => {
+            if (apiAuthenticator!.isValidAuthentication(password?.toString())) {
```

Comment:
Would it work without exclamation mark?
```suggestion
            if (apiAuthenticator.isValidAuthentication(password?.toString())) {
```

Reply:
Fixed